include/xtensor-io/xchunk_store_manager.hpp

```diff
@@ -138,6 +137,16 @@ namespace xt
         IP m_index_path;
     };
 
+    template <class EC, class IP, class EXT = empty_extension, class S>
```

Comment:
`EXT` should be the first template parameter, otherwise you would have to pass arguments for EC and IP when you pass an argument for `EXT`.

Reply:
But you always have to pass an argument for `EC` and `IP`, right? The way to create a chunked array with this factory function is e.g.:
```cpp
auto a = chunked_array<xfile_array<data_type, io_handler>, xzarr_index_path, xzarr_attrs>(shape, chunk_shape, path);
```

Reply:
Here `EC` is `xfile_array<data_type, io_handler>`, and `IP` is `xzarr_index_path`.

Reply:
OK, `IP` can have a default value, so you're right.

Reply:
Actually no, since `EC` is mandatory this can stay like this:
```cpp
template <class EC, class IP = xindex_path, class EXT = empty_extension, class S>
xchunked_array<xchunk_store_manager<EC, IP>, EXT> chunked_array(S&& shape, S&& chunk_shape, const std::string& path, std::size_t pool_size = 1);
```

Reply:
Ah indeed. This is not really convenient, maybe we should provide *more* factory methods with defaulted arguments. For instance:

```cpp
template <class T, class IOH, class EXT=empty_extension, class IP = xindex_path>
xchunked_array<xchunk_store_manager<xfile_array<T, IOH>, IP>, EXT>
chunked_file_array(args....);

template <class T, class IOH>
xchunked_array<xchunk_store_manager<xfile_array<T, IOH>, xzarr_index_path>, xzarr_attrs>
zarr_array(args....);

// Etc..
```

The implementation of these methods should rely on less specialized methods as much as possible.

Reply:
The problem is that an `xfile_array` as a default chunk container is not usable as-is, you still have to give it a path. Otherwise chunks will be stored in the current directory, probably colliding with chunks of other chunked arrays. I'm afraid we will have a lot of troubles if we allow this kind of default arguments.